_posts/2019-04-08-lecture-22.md

```diff
@@ -0,0 +1,339 @@
+---
+layout: distill
+title: "Lecture 22: Bayesian non-parametrics"
+description: An introduction to Bayesian non-parametrics and the Dirichlet process.
+date: 2019-04-08
+
+lecturers:
+  - name: Eric Xing
+    url: "https://www.cs.cmu.edu/~epxing/"
+
+authors:
+  - name: Yao Chong Lim  # author's full name
+    url: "#"  # optional URL to the author's homepage
+  - name: Yue Wu
+    url: "#"
+  - name: Dongshunyi Li
+    url: "#"
+  - name: Steve Landers
+    url: "#"
+
+editors:
+  - name: Maruan Al-Shedivat  # editor's full name
+    url: "#"  # optional URL to the editor's homepage
+
+---
+
+## Motivation
+
+### Clustering
+Bayesian non-parametrics is one a very popular topic in machine learning. The topic is not as popular right now but it is still useful since it's connected to a wide range of techniques we use today; for example, Gaussian Process can be applied to DNNs to make the layers infinitely wide.
+
+Let's start from a simpler problem (clustering). We already know a number of algorithms to solve this problem. Cluster analysis or clustering is the task of grouping a set of objects in such a way that objects in the same group (called a cluster) are more similar (in some sense) to each other than to those in other groups (clusters). Algorithms we've seen before (e.g. Kmeans, GMMs) solves if the number of groups were specified.
+
+However, in the real world, identifying the optimal number of clusters is still an open problem. In a two dimensional case, we can usually identify the number of clusters through direct visuallization, but it becomes hard to tell when the data is high-dimensional. Nevertheless, we can choose $k$ (number of groups) through some heuristics by comparing the results of different $k$ values, or compactness measure such as intra-cluster distances and inter-cluster distances, but these methods make the problem even harder by inducing an objective/measure that is different from the original object the algorithm tries to optimize.
+
+Moreover, when we are dealing with streaming data (e.g. new/media analysis), the clusters change every day. The algorirthm constantly needs to delete and add clusters. People often run Kmeans on the data every day to accomodate changes, but this induces "the mapping problem", since outputs of different Kmeans instances may not be aligned (i.e. Cluster 1 today might correspond to cluster 2 tomorrow).
```

Comment:
```suggestion
Moreover, when we are dealing with streaming data (e.g. new/media analysis), the clusters change every day. The algorithm constantly needs to delete and add clusters. People often run K-means on the data every day to accommodate changes, but this induces "the mapping problem", since outputs of different K-means instances may not be aligned (i.e. Cluster 1 today might correspond to cluster 2 tomorrow).
```